Objects/weakrefobject.c

```diff
@@ -29,6 +30,7 @@ init_weakref(PyWeakReference *self, PyObject *ob, PyObject *callback)
     self->wr_next = NULL;
     Py_XINCREF(callback);
     self->wr_callback = callback;
```

Comment:
While we are here...

```suggestion   
    self->wr_callback = Py_XNewRef(callback);
```